scope_benchmark_test.go

```diff
@@ -121,3 +122,82 @@ func BenchmarkHistogramExisting(b *testing.B) {
 		root.Histogram("foo", DefaultBuckets)
 	}
 }
+
+func benchmarkScopeReportingN(b *testing.B, numElems int) {
+	root, _ := NewRootScope(ScopeOptions{
+		Prefix:          "funkytown",
+		CachedReporter:  noopCachedReporter{},
+		SanitizeOptions: &alphanumericSanitizerOpts,
+	}, 0)
+	s := root.(*scope)
+
+	ids := make([]string, 0, numElems)
+	for i := 0; i < numElems; i++ {
+		id := fmt.Sprintf("take.me.to.%d", i)
+		ids = append(ids, id)
+		s.Counter(id)
+	}
+	b.ResetTimer()
+
+	for n := 0; n < b.N; n++ {
+		s.cachedReport()
+	}
+}
+
+func BenchmarkScopeReporting(b *testing.B) {
+	for i := 1; i <= 10000000; i *= 10 {
```

Comment:
10M is a bit overkill/unrealistic, limit to 1M so it completes faster?

Reply:
sure